### PR TITLE
Actually fix Lint CI

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
My suspicions were correct... it seems like the failures mainly (if not only) happen on a clean repository without build artifacts.
I've fixed it so it will work anyway - it's not pretty but I think it's all required

### Important Notes
None

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by somebody1234** [comment]
@wdanilo you pointed out it was verbose before... well, it's so much more verbose now DX

the alternative solution would be to remove all this specialcasing, and then just check for the existence of e.g. `build.json` and run a build if it's not there. thoughts?

https://github.com/enso-org/enso/blob/71ccc1d2fc276eac1b7892ec24704caa27516d86/app/ide-desktop/lib/types/modules.d.ts#L17-L45

**Comment by wdanilo** [comment]
> @wdanilo you pointed out it was verbose before... well, it's so much more verbose now DX
> 
> the alternative solution would be to remove all this specialcasing, and then just check for the existence of e.g. `build.json` and run a build if it's not there. thoughts?
> 
> https://github.com/enso-org/enso/blob/71ccc1d2fc276eac1b7892ec24704caa27516d86/app/ide-desktop/lib/types/modules.d.ts#L17-L45

Oh yes, let's do it as you propose please! Let's check for it, build it if it's not there and it'd be much better than this verbosity. Btw, why running build is needed like that? Our build script should build all dependencies automatically, so there should not be a situation when it's not there, right?

**Comment by somebody1234** [comment]
@wdanilo i think it's because lint doesnt depend on build. i'm not very familiar with the build CLI so maybe it'd be better for @mwu-tow to do it?

**Comment by somebody1234** [comment]
i think `build_ide` generates `build.json` and `linked-dist` (which are the exact things we need?)
however it seems like most of the defaults are coming from clap, so it's difficult to just build ide programmatically

File: app/ide-desktop/lib/types/modules.d.ts [hunk 1]
```diff
@@ -14,9 +15,35 @@ declare module '*/build.json' {
...
+    export namespace config {
+        export type Option<T> = import('runner/index').config.Option<T>
+        export const Option: typeof import('runner/index').config.Option
+        export type Group<
+            Options extends OptionsRecord,
+            Groups extends GroupsRecord
+        > = typeof import('runner/index').config.Group<Options, Groups>
+        export const Group: typeof import('runner/index').config.Group
+        export const objectToGroup: typeof import('runner/index').config.objectToGroup
+        export const objectToOption: typeof import('runner/index').config.objectToOption
```
Comment (wdanilo) [inline app/ide-desktop/lib/types/modules.d.ts:43]: this is terribly verbose and hard to maintain. I understand that the second PR is an alternative to this one, without this verbosity. Am I correct?

**Review by wdanilo — CHANGES_REQUESTED** [review]

File: app/ide-desktop/lib/types/modules.d.ts (hunk 1 above)
Comment (somebody1234) [reply app/ide-desktop/lib/types/modules.d.ts:43]: yeah - i definitely agree the other one is cleaner... and more importantly, doesnt require manually adding or removing exports

**Comment by somebody1234** [comment]
superseded by #6603

**Pull request closed without merging**